tests/test_07_l2vpn_return_codes.py

```diff
@@ -0,0 +1,256 @@
+import json
+import re
+import time
+from datetime import datetime, timedelta
+import uuid
+
+import pytest
+from random import randrange
+import requests
+
+from tests.helpers import NetworkTest
+
+SDX_CONTROLLER = 'http://sdx-controller:8080/SDX-Controller'
+
+class TestE2EReturnCodesEditL2vpn:
+    net = None
+
+    @classmethod
+    def setup_class(cls):
+        cls.net = NetworkTest(["ampath", "sax", "tenet"])
+        cls.net.wait_switches_connect()
+        cls.net.run_setup_topo()
+
+        # Create an L2VPN to edit later
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        cls.payload = {
+            "name": "Test L2VPN request",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "100"},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "100"}
+            ]
+        }
+        response = requests.post(api_url, json=cls.payload)
+        assert response.status_code == 201, response.text
+
+        response = requests.get(api_url)
+        data = response.json()
+        cls.key = list(data.keys())[0]
+
+    @classmethod
+    def teardown_class(cls):
+        cls.net.stop()
+   
+    @classmethod
+    def setup_method(cls):
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        response_json = response.json()
+        if len(response_json) == 0:
+        # Create an L2VPN to edit later
+            api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+            cls.payload = {
+                "name": "Test L2VPN request",
+                "endpoints": [
+                    {"port_id": "urn:sdx:port:ampath.net:Ampath3:50","vlan": "100"},
+                    {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "100"}
+                ]
+            }
+            response = requests.post(api_url, json=cls.payload)
+            assert response.status_code == 201, response.text
+
+        response = requests.get(api_url)
+        data = response.json()
+        cls.key = list(data.keys())[0]
```

Comment:
Not sure I understood the idea of using the setup_method here to obtain this L2VPN. This ware already saved from the setup_class, no?

Reply:
`setup_class` is called only once, at the beginning, but if any function deletes the l2vpn as part of testing, then other functions can fail. So, I implemented the `setup_method` function to avoid having to ensure that the l2vpn exists at the beginning of each function.